code/__HELPERS/unsorted.dm

```diff
@@ -1747,3 +1747,29 @@ Game Mode config tags:
 	ASSERT(Start.x < End.x && Start.y < End.y)
 
 	return block(Start, End) - block(locate(Start.x + 1, Start.y + 1, Start.z), locate(End.x - 1, End.y - 1, End.z))
+
+
+/proc/pick_rand_tele_turf(atom/hit_atom, var/inner_teleport_radius, var/outer_teleport_radius)
+	if((inner_teleport_radius < 1) || (outer_teleport_radius < inner_teleport_radius))
+		return 0
+
+	var/list/turfs = new/list()
+	var/turf/hit_turf = get_turf(hit_atom)
+	//This could likely use some standardization but I have no idea how to not break it.
+	for(var/turf/T in trange(outer_teleport_radius, hit_turf))
+		if(get_dist(T, hit_atom) <= inner_teleport_radius)
+			continue
+		if(is_blocked_turf(T) || istype(T, /turf/space))
+			continue
+		if(T.x > world.maxx-outer_teleport_radius || T.x < outer_teleport_radius)
+			continue
+		if(T.y > world.maxy-outer_teleport_radius || T.y < outer_teleport_radius)
+			continue
+		turfs += T
+	if(!turfs.len)
+		var/list/turfs_to_pick_from = list()
+		for(var/turf/T in trange(outer_teleport_radius, get_turf(hit_atom)))
+			if(get_dist(T, hit_atom) > inner_teleport_radius)
+				turfs_to_pick_from += T
+		turfs += pick(/turf in turfs_to_pick_from)
```

Comment:
I'll merge this as soon as someone explains to me how this works

Reply:
unless I'm misunderstanding what trange does, it seems self-explanatory: Randomly pick a turf that's between inner_teleport_radius and outer_teleport_radius tiles away from the tile the bluespace thing affects. ```/turf in turfs_to_pick_from``` could probably just lose the ```/turf in``` I suppose.

Reply:
It seems like this section is just the above section, but without the checks to see if the location picked is valid.